Restrictions/RangeInclusiveRestriction.py

```diff
@@ -0,0 +1,51 @@
+from typing import Optional
+
+from .BaseRestriction import *
+
+
+class RangeInclusiveRestriction(BaseRestriction):
+    def __init__(self, parent_restriction: BaseRestriction, restriction_name_in_xml: str, minimum: Optional = None,
+                 maximum: Optional = None):
+        super().__init__(restriction_name_in_xml)
+
+        self.parent_restriction = parent_restriction
+        self.minimum = minimum
+        self.maximum = maximum
+
+        if minimum is None and maximum is None:
+            raise Exception("ERROR: Illegal range, at least one of the minimum and maximum must be defined")
+
+        if minimum is not None and maximum is not None and maximum < minimum:
+            raise Exception("ERROR: Illegal range, maximum={} is less than minimum={}".format(maximum, minimum))
+
+    def value_matches_restriction(self, value: str) -> bool:
+        try:
+            value_converted = self.parent_restriction(value)
+        except RestrictionException as e:
+            return False
+
+        if self.minimum is None:
+            return value_converted <= self.maximum
+        elif self.maximum is None:
+            return self.maximum <= value_converted
```

Comment:
i think it should be return self.minimum <= value_converted. No?

Reply:
Your'e absolutely right